packages/base/Calendar/src/CalendarDay/utils/get-hover-and-focus-effects-classes.ts

```diff
@@ -0,0 +1,55 @@
+import type { DayState } from '../types'
+
+const applyToHoverAndFocus = (className: string) =>
+  `[&]:hover:${className} [&]:focus:${className}`
+
+export const getHoverAndFocusEffectsClasses = (state: DayState) => {
+  const {
+    isSelected,
+    isWeekend,
+    isDisabled,
+    isOutside,
+    isRangeEnd,
+    isRangeMiddle,
+  } = state
+
+  if (isDisabled || isSelected) {
+    return ''
+  }
+
+  if (isRangeEnd) {
+    // For Tailwind to pick classes up
+    // [&]:hover:text-white [&]:hover:bg-blue-500
+    // [&]:focus:text-white [&]:focus:bg-blue-500
+    return `${applyToHoverAndFocus('text-white')} ${applyToHoverAndFocus(
+      'bg-blue-500'
+    )}`
```

Comment:
If we are writing the classes on comments, we are now duplicating their value, both on the string value and the comment itself.
What is the benefit of this approach over just having the classname directly on the value?

Also note that because the comment is implicitly linked to the function implementation we need to keep the two in sync, if we change the function's implementation we need to remember to also change the comment where the function is being called. I would prefer to just have the class names inside the string constants
```suggestion
    return '[&]:hover:text-white [&]:hover:bg-blue-500 [&]:focus:text-white [&]:focus:bg-blue-500'
```

Reply:
good point ☝️ also for better readability it is nice to use array

```
const classes = [
  '[&]:hover:text-white',
  '[&]:hover:bg-blue-500',

  '[&]:focus:text-white',
  '[&]:focus:bg-blue-500',
]
```

as when there is the same prefix on each line your eyes will start ignoring it and you can better see the important values

Reply:
@augustobmoura @TomasSlama Agree, thank you guys! The change was reverted, now we deal with arrays which is slightly less readable, but instead we do not have to keep comments in sync (as they do not exist)